### PR TITLE
[simtests] Stress test committee changes

### PR DESCRIPTION
**Pull request by williampsmith**
As in title. The test starts with a committee size of 5, and a standby set of validator configs, of initial size 6, and at each epoch, requests to add 2 validators from the standby set, and requests to remove 2 validators from the current committee, until all validators in the standby set have been exhausted.

Note that we cannot test recycling the same validators in and out without more effort due to issues in fully shutting down all validator components when a node is demoted from a validator to a fullnode.

Note also that, due to the above, the test takes a long time to run, as, at each new epoch, we need to start new validators from genesis, and give time for them to catch up to the current epoch and reconfig as validators. This takes successively longer as we progress further from genesis. Currently we wait `30 sec * epochId`, but it's possible we can reduce this number somewhat.

### CODE REVIEW

File: crates/sui/tests/reconfiguration_tests.rs [hunk 1]
```diff
@@ -436,90 +439,400 @@ async fn test_reconfig_with_committee_change_basic() {
...
+
+    let mut new_node_config_clone = new_node_config.clone();
+    // Make sure that the new validator config shares the same genesis as the initial one.
+    new_node_config_clone.genesis = init_configs.validator_configs[0].genesis.clone();
+    let handle = start_node(
+        &new_node_config_clone,
+        RegistryService::new(Registry::new()),
+    )
+    .await;
+    // We have to manually insert the genesis objects since the test utility doesn't.
```
Comment (mystenmark) [inline crates/sui/tests/reconfiguration_tests.rs:477]: i think we can put the objects into genesis though, right?

Comment (williampsmith) [reply crates/sui/tests/reconfiguration_tests.rs:477]: I'll also use the same approach of writing to genesis for the test I introduced once I rebase. This was written before your PR had merged.

File: crates/sui/tests/reconfiguration_tests.rs [hunk 2]
```diff
@@ -436,90 +439,400 @@ async fn test_reconfig_with_committee_change_basic() {
...
+        trigger_reconfiguration(&authorities).await;
+
+        // Start new Nodes
+        let mut joined_auth_handles: Vec<_> = vec![];
+        for (_val, node_config) in joining_validators.clone() {
+            // Make sure that the new validator config shares the same genesis as the initial one.
+            let mut node_config = node_config.clone();
+            node_config.genesis = initial_network.validator_configs[0].genesis.clone();
+            let handle =
+                start_node(&node_config.clone(), RegistryService::new(Registry::new())).await;
```
Comment (lxfind) [inline crates/sui/tests/reconfiguration_tests.rs:658]: If you start all nodes from beginning, then the time takes for them to join would be much more predicable?

Comment (williampsmith) [reply crates/sui/tests/reconfiguration_tests.rs]: Done.

File: crates/sui/tests/reconfiguration_tests.rs (hunk 1 above)
Comment (williampsmith) [reply crates/sui/tests/reconfiguration_tests.rs]: This is rebased now and we're no longer doing this in my test.

File: crates/sui/tests/reconfiguration_tests.rs [hunk 3]
```diff
@@ -521,86 +522,434 @@ async fn test_reconfig_with_committee_change_basic() {
...
     .unwrap();
-    let transaction =
-        to_sender_signed_transaction(stake_tx_data, new_node_config.account_key_pair());
-    let effects = execute_transaction(&authorities, transaction)
-        .await
-        .unwrap();
-    assert!(effects.status().is_ok());
+    let transaction = to_sender_signed_transaction(stake_tx_data, node_config.account_key_pair());
+    let effects = execute_transaction(authorities, transaction).await.unwrap();
+    // assert!(effects.status().is_ok());
```
Comment (lxfind) [inline crates/sui/tests/reconfiguration_tests.rs]: Why not just the assert?

**Review by lxfind — APPROVED** [review]

File: crates/sui/tests/reconfiguration_tests.rs (hunk 3 above)
Comment (williampsmith) [reply crates/sui/tests/reconfiguration_tests.rs]: I actually just changed this back. I was printing the detailed info for debugging purposes. It's good now

**Pull request merged**